src/legacy/core_plugins/kibana/public/discover/np_ready/angular/discover.js

```diff
@@ -188,23 +177,118 @@ function discoverController(
   $scope,
   $timeout,
   $window,
-  AppState,
   Promise,
   config,
   kbnUrl,
   localStorage,
-  uiCapabilities,
-  getAppState,
-  globalState
+  uiCapabilities
 ) {
-  const filterStateManager = new FilterStateManager(globalState, getAppState, filterManager);
+  const subscriptions = new Subscription();
+  const $fetchObservable = new Subject();
+  let inspectorRequest;
+  const savedSearch = $route.current.locals.savedObjects.savedSearch;
+  $scope.searchSource = savedSearch.searchSource;
+  $scope.indexPattern = resolveIndexPatternLoading();
+  $scope.timeRangeObj = timefilter.getTime();
+  $scope.refreshInterval = timefilter.getRefreshInterval();
+
+  const getTimeField = () => {
```

Comment:
just to be sure indexPatternUtils.isDefault 

```TypeScript
export const isDefaultTypeIndexPattern = indexPattern => {
  // Default index patterns don't have `type` defined.
  return indexPattern.type == null;
};
```
Isn't that used to detect a rollup index pattern?

Reply:
Hmmm... Maybe the name `isDefault` is confusing?
Because indexPattern.isDefault sounds like we're checking if this is the index pattern marked as "default" in index pattern management .

Reply:
agreed, but it's hard to name this in a good way, at least the names that come to my mind are not good: isNoRollupIndexPattern, isBoringIndexPattern

maybe a comment would help, `there are x kinds of index pattern, apart from the default one, there is a rollup index pattern and ...`  do we have more?

Reply:
What about `isDefaultType`?

Reply:
would make sense!

Reply:
using it now as `isDefaultType` in discover